docker/scripts/functions.sh

```diff
@@ -98,6 +98,43 @@ function posttink() {
 	tink "POST" "${tink_host}" "${endpoint}" "${post_data}"
 }
 
+# returns a string of the BIOS vendor: "dell", "supermicro", or "unknown"
+function detect_bios_vendor() {
+	set +e
```

Comment:
This is because we're potentially running the dell/supermicro bios utilities on platforms which don't support them, so I'm trying to prevent the script from exiting/triggering autofail due to the non-zero exit status those commands will return.

Reply:
Thats ok though, you won't have bash exit because you handle all the error cases:
```
[manny@dellnix:~]$ bash <<EOF
set -e
if /opt/dell/srvadmin/bin/idracadm7; then                           
  echo we should not see this, there is no idracadm7
elif /opt/supermicro/sum/sum; then
  echo we should not see this, there is no sum tool
else
  echo ok so everything errored, its ok bash does not die because we handled it with if and else
fi
EOF
bash: line 2: /opt/dell/srvadmin/bin/idracadm7: No such file or directory
bash: line 4: /opt/supermicro/sum/sum: No such file or directory
ok so everything errored, its ok bash does not die because we handled it with if and else
```
This shouldn't trigger the exiting/autofail either.

Reply:
Ah, it is now since I'm no longer using intermediate variables to store/parse the output of the commands, which was my previous approach. Good catch.